src/main/scala/mrpowers/jodie/DeltaHelpers.scala

```diff
@@ -339,6 +339,65 @@ object DeltaHelpers {
       .save(storagePath)
   }
 
+  /**
+   * Validates and appends data to a Delta table. This function ensures that the provided DataFrame can be appended
+   * to the specified Delta table by checking data type compatibility and column presence.
+   *
+   * @param deltaTable   The Delta table to which data will be appended.
+   * @param appendDF     The DataFrame containing data to be appended.
+   * @param requiredCols The list of required columns in the appendDF.
+   * @param optionalCols The list of optional columns in the appendDF.
+   * @throws IllegalArgumentException if input arguments have an invalid type, are missing, or are empty.
+   * @throws IllegalArgumentException if required columns are missing in the provided Delta table.
+   * @throws IllegalArgumentException if a column in the append DataFrame is not part of the original Delta table.
+   */
+  def validateAppend(
+                      deltaTable: DeltaTable,
+                      appendDF: DataFrame,
+                      requiredCols: List[String],
+                      optionalCols: List[String]
+                    ): Unit = {
+    // Check if deltaTable is an instance of DeltaTable
+    if (!deltaTable.isInstanceOf[DeltaTable]) {
+      throw new IllegalArgumentException("An existing delta table must be specified.")
+    }
+
+    // Check if appendDF is an instance of DataFrame
+    if (!appendDF.isInstanceOf[DataFrame]) {
+      throw new IllegalArgumentException("You must provide a DataFrame that is to be appended.")
+    }
+
+    val appendDataColumns = appendDF.columns
+
+    // Check if all required columns are present in appendDF
+    for (requiredColumn <- requiredCols) {
+      if (!appendDataColumns.contains(requiredColumn)) {
+        throw new IllegalArgumentException(
```

Comment:
Instead of throwing an explicit exception you could use the precondition `require(boolean,message)`,  this is also idiomatic and frankly we should probably open an issue to change all the code to use this.